### PR TITLE
Mono-hop unidirectional payments

### PR DESCRIPTION
**Pull request by canndrew**
This is a work-in-progress PR for monodirectional, unihop (non-htlc) payments.

### CODE REVIEW

**Comment by knocte** [comment]
Please:
- rename any string that contains the substring GeewalletPayment to MonoHopUnidirectionalPayment.
- remove "WIP" prefix from main commit (because it passes the build, therefore it implements what is needed for no revocation yet; it's the revocation_and_ack which is WIP, AFAIU).
- revert the first commit about fsc.props to see if CI still passes without it.

**Comment by knocte** [comment]
Also comment out this line in ci.yml:
- https://github.com/nblockchain/DotNetLightning/blob/master/.github/workflows/ci.yml#L41
This way we will only push a BouncyCastle-based package. We need to rename the nupkg file too somehow (to not be named DotNetLightning; e.g. `DotNetLightning.Kiss`?)

**Comment by knocte** [comment]
I guess you can merge the commits here (except for the ones being proposed to Joe or the hacky one)

**Comment by knocte** [comment]
merge->squash

**Comment by knocte** [comment]
@canndrew i've updated the tip of this repo's master to sync with Joe's, you can rebase

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 1]
```diff
@@ -373,7 +384,8 @@ module internal OpenChannelMsgValidation =
             check
                 ourChannelReserve (<) msg.DustLimitSatoshis
                 "Dust limit too high for our channel reserve. our channel reserve is: %A . received dust_limit is: %A"
-        Validation.ofResult(check1) *^> check2 *^> check3
+        //Validation.ofResult(check1) *^> check2 *^> check3
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: commented code: please explain why it's left there (in a comment) or remove it

Comment (knocte) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: oops I see this is part of the hacky commit

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 2]
```diff
@@ -458,6 +470,18 @@ module internal AcceptChannelMsgValidation =
 
         (check1 |> Validation.ofResult) *^> check2 *^> check3 *^> check4 *^> check5 *^> check6 *^> check7
         
+module UpdateMonoHopUnidirectionalPaymentWithContext =
+    let internal checkWeHaveSufficientFunds (state: Commitments) (currentSpec) =
+        let fees = if (state.LocalParams.IsFunder) then (Transactions.commitTxFee (state.RemoteParams.DustLimitSatoshis) currentSpec) else Money.Zero
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: please split this long line in the same way we do in geewallet:
```
let foo =
    if bar then
        baz
    else
        bazz
```

Comment (knocte) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: (also no need for so many parenthesis)

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 3]
```diff
@@ -458,6 +470,18 @@ module internal AcceptChannelMsgValidation =
...
+        let missing = currentSpec.ToRemote.ToMoney() - state.RemoteParams.ChannelReserveSatoshis - fees
+        if (missing < Money.Zero) then
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: nit: no need for parenthesis

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 4]
```diff
@@ -458,6 +470,18 @@ module internal AcceptChannelMsgValidation =
...
+            sprintf "We don't have sufficient funds to send mono-hop unidirectional payment. current to_remote amount is: %A. Remote Channel Reserve is: %A. and fee is %A"
+                    (currentSpec.ToRemote.ToMoney())
+                    (state.RemoteParams.ChannelReserveSatoshis)
+                    (fees)
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: no need for parenthesis in the above 2 lines (the 3rd line needs them though)

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 5]
```diff
@@ -472,7 +496,23 @@ module UpdateAddHTLCValidation =
...
-    
+module internal MonoHopUnidirectionalPaymentValidationWithContext =
+    let checkWeHaveSufficientFunds (state: Commitments) (currentSpec) =
+        let fees =
+            if state.LocalParams.IsFunder then
+                Transactions.commitTxFee state.RemoteParams.DustLimitSatoshis currentSpec
+            else
+                Money.Zero
+        let missing = currentSpec.ToRemote.ToMoney() - state.RemoteParams.ChannelReserveSatoshis - fees
+        if (missing < Money.Zero) then
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: nit: no need for parenthesis above

File: src/DotNetLightning.Core/Channel/Channel.fs [hunk 6]
```diff
@@ -388,6 +388,32 @@ module Channel =
...
+                let payment: MonoHopUnidirectionalPayment = {
+                    ChannelId = state.Commitments.ChannelId
+                    Amount = cmd.Amount
+                }
+                let commitments1 = state.Commitments.AddLocalProposal(payment)
+
+                let remoteCommit1 =
+                    match commitments1.RemoteNextCommitInfo with
+                    | RemoteNextCommitInfo.Waiting info -> info.NextRemoteCommit
+                    | RemoteNextCommitInfo.Revoked _info -> commitments1.RemoteCommit
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/Channel.fs:405]: as revokation is not implemented yet, this line above is not needed right?

File: tests/DotNetLightning.Infrastructure.Tests/ChannelOperationTests.fs [hunk 7]
```diff
@@ -312,8 +337,164 @@ let tests =
...
+                ""
+            Expect.equal
+                aliceExpectedAmount
+                bobCommitments.RemoteCommit.Spec.ToLocal
+                ""
+            Expect.equal
+                bobExpectedAmount
+                bobCommitments.RemoteCommit.Spec.ToRemote
+                ""
+            *)
```
Comment (knocte) [inline tests/DotNetLightning.Infrastructure.Tests/ChannelOperationTests.fs]: can you please create a "revokation WIP PR" where you can add this piece of code above^ (and the `Send mono-hop unidirectional payment with revoke_and_ack` test), to remove it from here

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -21,22 +21,11 @@ jobs:
...
+    - name: Upload BouncyCastle nuget package
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
-        if [ ${{ secrets.NUGET_API_KEY }} ] && [ $GITHUB_REF == "refs/heads/master" ]; then
-            dotnet nuget push ./bin/Release/DotNetLightning.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
-            dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+        if [ ${{ secrets.NUGET_API_KEY }} ]; then
+            dotnet nuget push ./bin/Release/DotNetLightning.Kiss.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+            #dotnet nuget push ./bin/Release/DotNetLightning.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+            #dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
```
Comment (knocte) [inline .github/workflows/ci.yml]: let's remove the commented code lines here

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -21,22 +21,11 @@ jobs:
...
-    - name: Clean to prepare for NSec build
-      run: |
-        dotnet clean
-    - name: Run tests
-      run: |
-        dotnet test
-    - name: Package NSec build
-      run: |
-        cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
-        dotnet pack -p:Configuration=Release -p:Version=1.1.0-date`date +%Y%m%d-%H%M`.git-`echo $GITHUB_SHA | cut -c 1-7`
```
Comment (knocte) [inline .github/workflows/ci.yml]: is there a way to place a multi-line comment in YAML?

File: src/DotNetLightning.Core/Channel/Channel.fs (hunk 6 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Channel.fs:405]: We still need to use the correct remote commitment though. So changing this to always use `commitments1.RemoteCommit` would break it.

Comment (knocte) [reply src/DotNetLightning.Core/Channel/Channel.fs:405]: sorry I don't understand

File: src/DotNetLightning.Core/Channel/Channel.fs [hunk 10]
```diff
@@ -730,8 +755,12 @@ module Channel =
             { c with State = ChannelState.Normal data }
 
         // ----- normal operation --------
+        | WeAcceptedCMDMonoHopUnidirectionalPayment(_, newCommitments), ChannelState.Normal d ->
+            { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
         | WeAcceptedCMDAddHTLC(_, newCommitments), ChannelState.Normal d ->
             { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
+        | WeAcceptedMonoHopUnidirectionalPayment(newCommitments), ChannelState.Normal d ->
+            { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/Channel.fs]: let's avoid 1-character variable names please

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -21,22 +21,9 @@ jobs:
...
-    - name: Upload nuget packages (BouncyCastle and native)
+    - name: Upload BouncyCastle nuget package
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
-        if [ ${{ secrets.NUGET_API_KEY }} ] && [ $GITHUB_REF == "refs/heads/master" ]; then
-            dotnet nuget push ./bin/Release/DotNetLightning.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
-            dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+        if [ ${{ secrets.NUGET_API_KEY }} ]; then
```
Comment (knocte) [inline .github/workflows/ci.yml]: you need to restore the `&& [ $GITHUB_REF == "refs/heads/master" ]` bit here before we can merge this in our fork

File: src/DotNetLightning.Core/Channel/ChannelError.fs (hunk 1 above)
Comment (knocte) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: thanks for adding the comment, but anyway I don't think this hunk should be merged in our fork (it's fine that your earlier versions of .Kiss include it, but this will be removed before we merge this PR)

File: src/DotNetLightning.Core/Channel/Channel.fs [hunk 12]
```diff
@@ -676,104 +701,108 @@ module Channel =
...
-        | WeAcceptedUpdateFee(_msg), ChannelState.Normal _d -> c
-
-        | WeAcceptedCMDSign(_msg, newCommitments), ChannelState.Normal d ->
-            { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
-        | WeAcceptedCommitmentSigned(_msg, newCommitments), ChannelState.Normal d ->
-            { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
-
-        | WeAcceptedRevokeAndACK(newCommitments), ChannelState.Normal d ->
-            { c with State = ChannelState.Normal({ d with Commitments = newCommitments }) }
+        | WeAcceptedCMDMonoHopUnidirectionalPayment(_, newCommitments), ChannelState.Normal stateData ->
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/Channel.fs]: @canndrew I only meant for renaming of variables on new code, not on DNL code that is already there, we want our delta between our fork and Joe's DNL to be as minimal as possible

Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Channel.fs]: Ah. I can change it back except for the renaming of `d => stateData` then.

File: src/DotNetLightning.Core/Channel/Channel.fs (hunk 6 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Channel.fs:405]: Revocation isn't implemented in the sense that we don't validate or save the revocation secrets that the peer sends us. We still send and receive `revoke_and_ack` messages though as part of the process of updating the commitments.

File: tests/DotNetLightning.Infrastructure.Tests/ChannelOperationTests.fs (hunk 7 above)
Comment (canndrew) [reply tests/DotNetLightning.Infrastructure.Tests/ChannelOperationTests.fs]: This wasn't actually necessary since the test now checks that both peers send the `revoke_and_ack`, so the highly-commented test case was redundant and has been removed.

Checking that revocation secrets are handled correctly would require an entirely different test.

File: .github/workflows/ci.yml (hunk 9 above)
Comment (canndrew) [reply .github/workflows/ci.yml]: No. Some googling suggests that only single-line comments are supported.

Comment (knocte) [reply .github/workflows/ci.yml]: alright :(

**Comment by knocte** [comment]
So should we squash all commits here except for the HACKYCOMMIT and any other change that shouldn't get merged in our fork?

File: src/DotNetLightning.Core/Channel/Channel.fs (hunk 6 above)
Comment (knocte) [reply src/DotNetLightning.Core/Channel/Channel.fs:405]: why do we need to send those messages if we don't do revokation?

File: src/DotNetLightning.Core/Channel/Channel.fs (hunk 12 above)
Comment (knocte) [reply src/DotNetLightning.Core/Channel/Channel.fs]: yes please

**Comment by canndrew** [comment]
Sure. I might try to send the last commit (Make GetFailureMsgData format non-printable error messages) upstream though.

**Comment by knocte** [comment]
>  might try to send the last commit (Make GetFailureMsgData format non-printable error messages) upstream though.

ok then separate that one but let's not delay this so much

File: .github/workflows/ci.yml [hunk 13]
```diff
@@ -21,22 +21,12 @@ jobs:
...
-    - name: Upload nuget packages (BouncyCastle and native)
+    - name: Upload BouncyCastle nuget package
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
-        if [ ${{ secrets.NUGET_API_KEY }} ] && [ $GITHUB_REF == "refs/heads/master" ]; then
-            dotnet nuget push ./bin/Release/DotNetLightning.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
-            dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+        # NOTE:
+        # the geewallet-payments branch reference here can be removed once this
+        # is merged to master
```
Comment (knocte) [inline .github/workflows/ci.yml]: not "once this is merged to master"; this has to be removed before we merge it to master

File: src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs [hunk 14]
```diff
@@ -1477,3 +1484,19 @@ type GossipTimestampFilter = {
             ls.Write(this.FirstTimestamp, false)
             ls.Write(this.TimestampRange, false)
             
+[<CLIMutable>]
+type MonoHopUnidirectionalPayment = {
+    mutable ChannelId: ChannelId
+    mutable Amount: LNMoney
```
Comment (knocte) [inline src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs:1556]: is mutable really needed here???

Comment (canndrew) [reply src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs:1556]: All the other message types have it, though I don't know why. Removing it gives a pile of compilation errors.

File: .github/workflows/ci.yml (hunk 13 above)
Comment (canndrew) [reply .github/workflows/ci.yml]: I've removed it now.

File: src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs (hunk 14 above)
Comment (knocte) [reply src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs:1556]: i guess the real way to fix this is change `ILightningSerializable<'T>`'s Deserialize method to return `ILightningSerializable<'T>` instead of unit, right?

File: src/DotNetLightning.Core/Channel/Channel.fs (hunk 6 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Channel.fs:405]: The protocol specifies that a peer must always reply with a `revoke_and_ack` when it receives a `commitment_signed`. DNL uses the receiving of the `revoke_and_ack` to progress to the next commitments state. It would have been harder (ie. required more changes to DNL) to remove the revoke_and_ack, and would have been a step backwards since we'll need it eventually anyway for when we add proper revocation handling.

File: src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs (hunk 14 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs:1556]: I think that would be sufficient, since I can't see anywhere else in the code where the fields are being mutated. Returning a value rather than mutating an argument would be more "functional" too.

Comment (knocte) [reply src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs:1556]: ok I've added an issue about this: https://github.com/joemphilips/DotNetLightning/issues/87

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 15]
```diff
@@ -458,6 +469,22 @@ module internal AcceptChannelMsgValidation =
 
         (check1 |> Validation.ofResult) *^> check2 *^> check3 *^> check4 *^> check5 *^> check6 *^> check7
         
+module UpdateMonoHopUnidirectionalPaymentWithContext =
+    let internal checkWeHaveSufficientFunds (state: Commitments) (currentSpec) =
+        let fees =
+            if state.LocalParams.IsFunder then
+                Transactions.commitTxFee (state.RemoteParams.DustLimitSatoshis) currentSpec
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: no need for parenthesis above

File: src/DotNetLightning.Core/Utils/LNMoney.fs [hunk 16]
```diff
@@ -37,6 +37,8 @@ type LNMoney = | LNMoney of int64 with
         let satoshi = Checked.op_Multiply (amount) (decimal lnUnit)
         LNMoney(Checked.int64 satoshi)
 
+    static member FromMoney (money: Money) =
+        LNMoney.Satoshis(money.Satoshi)
```
Comment (knocte) [inline src/DotNetLightning.Core/Utils/LNMoney.fs]: @canndrew no need for parenthesis here

File: src/DotNetLightning.Core/Channel/ChannelTypes.fs [hunk 17]
```diff
@@ -397,3 +397,70 @@ type ChannelState =
...
+                        |> LNMoney.FromMoney
+                    else
+                        LNMoney.Zero
+                let channelReserve =
+                    commitments.RemoteParams.ChannelReserveSatoshis
+                    |> LNMoney.FromMoney
+                let totalBalance = reduced.ToRemote
+                let untrimmedSpendableBalance = totalBalance - channelReserve - fees
+                let htlcSuccessFee =
+                    reduced.FeeRatePerKw.ToFee(Transactions.Constants.HTLC_SUCCESS_WEIGHT)
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelTypes.fs]: @canndrew no need for parenthesis here either

File: src/DotNetLightning.Core/Channel/ChannelTypes.fs [hunk 18]
```diff
@@ -397,3 +397,70 @@ type ChannelState =
...
+                    |> LNMoney.FromMoney
+                let htlcFee =
+                    reduced.FeeRatePerKw.ToFee(Transactions.Constants.COMMITMENT_TX_WEIGHT_PER_HTLC)
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelTypes.fs]: @canndrew no need for parenthesis here either

**Comment by knocte** [comment]
@canndrew pls rebase when u have time

File: .github/workflows/publish_master.yml [hunk 19]
```diff
@@ -38,4 +38,4 @@ jobs:
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
         dotnet pack . -p:Configuration=Release --version-suffix date`date +%Y%m%d-%H%M`-git-`echo $GITHUB_SHA | cut -c 1-7`-${{ matrix.RID }}
-        dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
\ No newline at end of file
+        dotnet nuget push ./bin/Release/DotNetLightning.Kiss.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
```
Comment (knocte) [inline .github/workflows/publish_master.yml:35]: this looks wrong, we should push BouncyCastle binaries, not native

File: src/DotNetLightning.Core/Channel/ChannelError.fs (hunk 15 above)
Comment (knocte) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: @canndrew ping ^ (nit)

File: src/DotNetLightning.Core/Channel/Commitments.fs [hunk 20]
```diff
@@ -183,3 +183,46 @@ type Commitments = {
...
+                    |> LNMoney.FromMoney
+                else
+                    LNMoney.Zero
+            let channelReserve =
+                this.RemoteParams.ChannelReserveSatoshis
+                |> LNMoney.FromMoney
+            let totalBalance = reduced.ToRemote
+            let untrimmedSpendableBalance = totalBalance - channelReserve - fees
+            let htlcSuccessFee =
+                reduced.FeeRatePerKw.ToFee(Transactions.Constants.HTLC_SUCCESS_WEIGHT)
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/Commitments.fs]: why do we use any HTLC-related stuff here if we don't use HTLCs?

File: src/DotNetLightning.Core/Serialize/Features.fs [hunk 21]
```diff
@@ -194,9 +199,7 @@ type FeatureBit private (bitArray) =
             else
                 return (FeatureBit(ba))
         }
-    static member Zero =
-        let b: bool array = [||]
-        b |> BitArray |> FeatureBit
+    static member Zero = FeatureBit()
```
Comment (knocte) [inline src/DotNetLightning.Core/Serialize/Features.fs]: is this change really needed for this PR?

File: src/DotNetLightning.Core/Utils/LNMoney.fs (hunk 16 above)
Comment (knocte) [reply src/DotNetLightning.Core/Utils/LNMoney.fs]: @canndrew ping

File: src/DotNetLightning.Core/Channel/Commitments.fs (hunk 20 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Commitments.fs]: Because I had to think hard about how exactly this function needs to be implemented and I didn't want to have to do that again, or to have the code silently break, when we eventually add support for HTLCs. So I just made it HTLC-compatible to begin with.

File: src/DotNetLightning.Core/Serialize/Features.fs (hunk 21 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Serialize/Features.fs]: Nope. I can remove it.

File: .github/workflows/publish_master.yml (hunk 19 above)
Comment (canndrew) [reply .github/workflows/publish_master.yml:35]: Oops. I've fixed it so that we only upload the BouncyCastle build and we upload it as DotNetLightning.Kiss

File: src/DotNetLightning.Core/Channel/Commitments.fs (hunk 20 above)
Comment (knocte) [reply src/DotNetLightning.Core/Channel/Commitments.fs]: that's great but I'd prefer if you move this code into another PR, to pick it up later

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 22]
```diff
@@ -145,6 +147,15 @@ and InvalidAcceptChannelError = {
         Errors = e
     }
     
+and InvalidMonoHopUnidirectionalPaymentError = {
+    Msg: MonoHopUnidirectionalPayment
+    Errors: string list
+}
+    with
+    static member Create msg e = {
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs]: hey let's remove this Create method, as the record can be instantiated easily anyway without it

Comment (canndrew) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: I can do. All the other invalid msg types have it though, though they all seem equally pointless. Is it important to be consistent here?

Comment (knocte) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: good point; let's remove it here and I'll prepare a PR to remove them upstream

Comment (canndrew) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: Done.

File: src/DotNetLightning.Core/Channel/Commitments.fs (hunk 20 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/Commitments.fs]: I've removed the HTLC fees from the calculation and created a seperate PR which adds them back here: #3

File: src/DotNetLightning.Core/Channel/ChannelError.fs (hunk 15 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/ChannelError.fs]: Fixed.

File: src/DotNetLightning.Core/Channel/ChannelTypes.fs (hunk 17 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/ChannelTypes.fs]: Fixed.

File: src/DotNetLightning.Core/Utils/LNMoney.fs (hunk 16 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Utils/LNMoney.fs]: Fixed.

File: src/DotNetLightning.Core/Channel/ChannelTypes.fs (hunk 18 above)
Comment (canndrew) [reply src/DotNetLightning.Core/Channel/ChannelTypes.fs]: Fixed.
